File: python/Car.py
from Vehicle import Vehicle


class Car(Vehicle):
    __jumlah_kursi = ""
    __jumlah_pintu = ""

    # construktor
    def __init__(self, jumlah_kursi, jumlah_pintu, plat, merk, tahun, warna):
        # mengisi construktor parent didalam contructor anak (mobil)
        super().__init__(plat, merk, tahun, warna)
        self.__jumlah_kursi = jumlah_kursi
        self.__jumlah_pintu = jumlah_pintu

    # getter setter

    def getJumlahKursi(self):
        return self.__jumlah_kursi

    def setJumlahKursi(self, jumlah):
        self.__jumlah_kursi = jumlah

    def getJumlahPintu(self):
        return self.__jumlah_pintu

    def setJumlahPintu(self, jumlah):
        self.__jumlah_pintu = jumlah

File: cpp/Car.cpp
#pragma once
#include <iostream>
#include <string>

#include "Vehicle.cpp"

using namespace std;

class Car : public Vehicle {
 private:
  string jumlah_pintu;
  string jumlah_kursi;

 public:
  Car() {}
  Car(string jumlah_pintu, string jumlah_kursi) {
    this->jumlah_pintu = jumlah_pintu;
    this->jumlah_kursi = jumlah_kursi;
  }
  // constructor : overloading (delegation)
  /*
      disaat constructor yang  berisi atribut Vehicle dipanggil, maka nanti
     atribut data yang terdapat disini bakal di oper ke construktor yang ada di
     Vehicle

      Contructor Car --> manggil Constructor Vehicle (passing atribut nya juga)
  */
  Car(string jumlah_pintu, string jumlah_kursi, string plat, string merk,
      string tahun, string warna)
      : Vehicle(plat, merk, tahun, warna) {
    this->jumlah_pintu = jumlah_pintu;
    this->jumlah_kursi = jumlah_kursi;
  }

  // setter getter
  /*
      Getter is Accessor --> tempat mengakses
      Setter is Mutator --> tempat mengganti
  */
  string getJumlahPintu() { return this->jumlah_pintu; }
  void setJumlahpintu(string jumlahPintu) { this->jumlah_pintu = jumlahPintu; }

  string getJumlahKursi() { return this->jumlah_kursi; }
  void setJumlahKursi(string jumlahKursi) { this->jumlah_kursi = jumlahKursi; }

  void printDetails() override {
    // kita pake polymorphism
    Vehicle::printDetails();
    cout << "Jumlah Pintu : " << getJumlahPintu() << endl;
    cout << "Jumlah Kursi : " << getJumlahKursi() << endl;
  }

  ~Car() {}
};